website/docs/language/modules/develop/refactoring.html.md

```diff
@@ -0,0 +1,461 @@
+---
+layout: "language"
+page_title: "Refactoring"
+sidebar_current: "docs-modules-recactoring"
+description: |-
+  How to make backward-compatible changes to modules already in use.
+---
+
+# Refactoring
+
+~> **Note:** Explicit refactoring declarations with `moved` blocks will work
+only in Terraform v1.1 and later. When working with earlier Terraform versions,
+or for refactoring actions too complex to express as `moved` blocks, you could
+use
+[the `terraform state mv` CLI command](/docs/cli/commands/state/mv.html)
+as a separate step instead.
+
+In shared modules and long-lived configurations, you may eventually outgrow
+your initial module structure and resource names. For example, you might decide
+that what was previously one child module makes more sense as two separate
+modules and move a subset of the existing resources to the new one.
+
+Terraform compares previous state with new configuration, correlating by
+each module or resource's unique address. Therefore _by default_ Terraform will
+understand moving or renaming an object as an intent to destroy the object
+at the old address and to create a new object at the new address.
```

Comment:
You write so well, Martin, that I'm nitpicking now :p But, I try to avoid saying "will." Often we can just make a stronger, simpler statement that means the same thing by just omitting it. 

```suggestion
Terraform compares previous state with new configuration, correlating by each module or resource's unique address. Therefore _by default_ Terraform understands moving or renaming an object as an intent to destroy the object at the old address and create a new object at the new address.
```